src/service/auth.service.ts

```diff
@@ -0,0 +1,70 @@
+import {User} from '../models'
+import sendEmail from '../helpers/sendEmail'
+import crypto from 'crypto'
+import { Op } from 'sequelize'
+import { resetPasswordTemplate } from '../helpers/EmailTemplates/resetPasswordTemplate'
+
+class AuthService {
+  public static async querySingleUser(options: any): Promise<User> {
+    const user = await User.findOne({ where: options })
+    if (!user) {
+      throw new Error('User not found')
+    }
+    return user
+  }
+  public async forgotPassword(email: string) {
+    const user = await AuthService.querySingleUser({ email })
+    const resetToken = crypto.randomBytes(32).toString('hex')
+    const resetTokenExpiration = new Date(
+      Date.now() + 24 * 60 * 60 * 1000,
+    ).toISOString()
+
+    user.resetToken = resetToken
+    user.resetTokenExpiration = resetTokenExpiration
+    await user.save()
+
+    // Create reset password link
+    const resetUrl = `${process.env.DEPLOYED_URL}/api/v1/auth/reset-password/${resetToken}`
+
+    // Generate email content (HTML format recommended for better formatting)
+    const emailContent = resetPasswordTemplate(resetUrl)
+
+    await sendEmail(user.email, 'Password Reset', emailContent) // Send email using sendEmail function
+
+    return 'A password reset link has been sent to your email'
+  }
+
+  async resetPassword(token: string, password: string) {
```

Comment:
Let's add static to all methods. So we don't have to create instances of the class every time we want to access a method. Apply this to all methods
```suggestion
  public static async resetPassword(token: string, password: string) {
```